twitter/twitter_client.bal

```diff
@@ -0,0 +1,1057 @@
+//
+// Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+//
+
+import ballerina/http;
+import ballerina/io;
+
+string clientIdStr = "";
+string clientSecretStr = "";
+string accessTokenStr = "";
+string accessTokenSecretStr = "";
+
+# Twitter Client object.
+#
+# + httpClient - HTTP Client.
+# + userClient - Twitter user client.
+# + trendsClient - Twitter trends client.
+public type TwitterClient client object {
+
+    http:Client httpClient;
+    TwitterUserClient|error userClient;
+    TwitterTrendsClient|error trendsClient;
+
+    public function __init(TwitterConfiguration twitterConfig) returns error? {
+        self.httpClient = new(TWITTER_API_URL, config = twitterConfig.clientConfig);
+        var isValid = verifyCredentials(self.httpClient, twitterConfig.clientId, twitterConfig.clientSecret,
+            twitterConfig.accessToken, twitterConfig.accessTokenSecret);
+        if (isValid is error) {
+            return isValid;
+        }
+        self.userClient = new(self);
+        self.trendsClient = new(self);
+    }
+
+    # Get the user client to invoke user related functions.
+    #
+    # + return - Returns `TwitterUserClient` object.
+    public function getUserClient() returns TwitterUserClient? {
+        TwitterUserClient|error userClient = self.userClient;
+        if (userClient is TwitterUserClient) {
+            return userClient;
+        }
+        return ();
```

Comment:
Once we change userClient to not be of a union type, we can restrict the return type to just `TwitterUserClient` here and simply return self.userClient, which IMO will be cleaner.

Same applies for `trendsClient` too.

Comments for this implementation, even though they might no longer be valid once we change the variable types:
- We don't need to explicitly say `return ();`. Not having a return statement at the end here is the same as explicitly saying `return ();` here.
- If an error was set, IMO we should return the error itself, returning `()` just increases the requirements on the part of the user, where they would have to have the understanding that returning `()` indicates that an error occurred.